### PR TITLE
Fix CustomAGGrid:postSortRows : optimize groupPostSort

### PR DESCRIPTION
**Pull request by sBouzols**
Optimize groupPostSort N² complexity to N.

### CODE REVIEW

**Review by EstherDarkish — APPROVED** [review]
Code OK Tests OK

File: src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx [hunk 1]
```diff
@@ -114,19 +114,28 @@ const ShortCircuitAnalysisResult: FunctionComponent<
...
-        const idRows = sortedRows.filter((row) => row.data[idField] != null);
-        idRows.forEach((idRow) => {
-            result.push(idRow);
-            result.push(
-                ...sortedRows.filter(
-                    (row) => row.data[linkedIdField] === idRow.data[idField]
-                )
-            );
+        const rowsMap = new Map<string, IRowNode[]>();
+        // first reduce by main resource idField
```
Comment (jonenst) [inline src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx:118]: Reduce => index

File: src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx [hunk 2]
```diff
@@ -114,19 +114,28 @@ const ShortCircuitAnalysisResult: FunctionComponent<
...
+        sortedRows.forEach((row) => {
+            if (row.data[idField] != null) {
+                rowsMap.set(row.data[idField], [row]);
+            }
         });
 
-        return result;
+        // then reduce by linked resource linkedIdField
```
Comment (jonenst) [inline src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx:125]: Index

File: src/components/security-analysis-result.js [hunk 3]
```diff
@@ -257,29 +257,40 @@ const SecurityAnalysisResult = ({ onClickNmKConstraint, result }) => {
...
-        idRows.forEach((idRow) => {
-            //add group's parent first
-            result.push(idRow);
-            //then add all elements which belongs to this group
-            result.push(
-                ...sortedRows.filter(
-                    (row) => row.data[linkedElementId] === idRow.data[idField]
-                )
-            );
+        // first reduce by main resource idField
```
Comment (jonenst) [inline src/components/security-analysis-result.js:265]: Index

File: src/components/security-analysis-result.js [hunk 4]
```diff
@@ -257,29 +257,40 @@ const SecurityAnalysisResult = ({ onClickNmKConstraint, result }) => {
...
+        sortedRows.forEach((row) => {
+            if (row.data[idField] != null) {
+                rowsMap.set(row.data[idField], [row]);
+            }
         });
 
-        return result;
+        // then reduce by linked resource linkedElementId
```
Comment (jonenst) [inline src/components/security-analysis-result.js:272]: Index

File: src/components/security-analysis-result.js [hunk 5]
```diff
@@ -257,29 +257,40 @@ const SecurityAnalysisResult = ({ onClickNmKConstraint, result }) => {
         linkedElementId,
         isContingency
     ) => {
-        const result = [];
-        //get all groups ids
-        const idRows = sortedRows.filter((row) => row.data[idField] != null);
+        // Because Map remembers the original insertion order of the keys.
```
Comment (jonenst) [inline src/components/security-analysis-result.js:260]: Put this comment in the other code

**Review by archendel — APPROVED** [review]

File: src/components/security-analysis-result.js (hunk 5 above)
Comment (sBouzols) [reply src/components/security-analysis-result.js:260]: done

File: src/components/security-analysis-result.js (hunk 4 above)
Comment (sBouzols) [reply src/components/security-analysis-result.js]: done

File: src/components/security-analysis-result.js (hunk 3 above)
Comment (sBouzols) [reply src/components/security-analysis-result.js]: done

File: src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx (hunk 2 above)
Comment (sBouzols) [reply src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx]: done

File: src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx (hunk 1 above)
Comment (sBouzols) [reply src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx]: done

**Pull request merged**